contracts/BatchZkSyncDeposit.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.6.8;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @dev We use ABIEncoderV2 to enable encoding/decoding of the array of structs. The pragma
+ * is required, but ABIEncoderV2 is no longer considered experimental as of Solidity 0.6.0
+ */
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @notice Interface for the zkSync contract
+ */
+interface IZkSync {
+  function depositETH(address _franklinAddr) external payable;
+
+  function depositERC20(
+    IERC20 _token,
+    uint104 _amount,
+    address _franklinAddr
+  ) external;
+}
+
+/**
+ * @notice Enables batch deposits of multiple tokens into the zkSync contract with one transaction
+ */
+contract BatchZkSyncDeposit is Ownable, Pausable, ReentrancyGuard {
+  using SafeERC20 for IERC20;
+
+  // Placeholder token address to represent ETH deposits
+  IERC20 public constant ETH_TOKEN_PLACHOLDER = IERC20(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
```

Comment:
Typo: `ETH_TOKEN_PLACHOLDER` should be `ETH_TOKEN_PLACEHOLDER`

Also, it feels ab it awkward declaring this an IERC20 when it's not, though I see why you did it. Why does this need to be public, though? Thinking of some external contract trying to use it (for some reason) as an IERC20.

Reply:
I don't think it rally makes a difference whether it's public or private. We neglected to specify a visibility for this in BulkCheckout, and it seems that it's private by default in that case. So for consistency I've made it private and fixed the typo in dcacc42

Reply:
Sounds good 👍